dbus/methods.go

```diff
@@ -196,7 +196,21 @@ func (c *Conn) KillUnit(name string, signal int32) {
 
 // KillUnitContext same as KillUnit with context
 func (c *Conn) KillUnitContext(ctx context.Context, name string, signal int32) {
-	c.sysobj.CallWithContext(ctx, "org.freedesktop.systemd1.Manager.KillUnit", 0, name, "all", signal).Store()
+	c.KillSpecificProcessUnitContext(ctx, name, All, signal)
+}
+
+type Who string
+
+const (
+	All     Who = "all"
+	Main    Who = "main"
+	Control Who = "control"
+)
+
+// KillSpecificProcessUnitContext is like KillUnitContext, but allows you to specify which process in the unit to send
+// the signal to.
+func (c *Conn) KillSpecificProcessUnitContext(ctx context.Context, name string, who Who, signal int32) error {
```

Comment:
Bikeshedding: this a bit of a mouthful, and I think for new APIs we can avoid the `Context` suffix. Maybe `KillUnitWithTarget` and then rename the argument as `target Who`?